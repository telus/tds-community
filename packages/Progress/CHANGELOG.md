# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [2.1.0](https://github.com/telusdigital/tds-community/compare/@tds/community-progress@2.0.1...@tds/community-progress@2.1.0) (2021-09-21)


### Features

* **community-progress:** react 17 and styled components 5 ([d44cfb8](https://github.com/telusdigital/tds-community/commit/d44cfb81a3509f86adeb3271c0d3a1b3918fbf85))





## [2.0.1](https://github.com/telusdigital/tds-community/compare/@tds/community-progress@2.0.0...@tds/community-progress@2.0.1) (2020-09-09)


### Bug Fixes

* **community-progress:** update util-helpers version ([c87783c](https://github.com/telusdigital/tds-community/commit/c87783cfee95f1774a6afec938af5dc8161ac4f4))





# [2.0.0](https://github.com/telusdigital/tds-community/compare/@tds/community-progress@1.0.2...@tds/community-progress@2.0.0) (2020-02-07)


### Features

* **community-progress:** implement new design standards ([7eaf907](https://github.com/telusdigital/tds-community/commit/7eaf907))


### BREAKING CHANGES

* **community-progress:** progress bar height has been adjusted and no longer changes with viewport
* **community-progress:** allyLabel prop in Progress is required to aid assistive technology
* **community-progress:** 'primary' variant is now named 'positive'
* **community-progress:** 'error' variant is now named 'negative'
* **community-progress:** removed 'secondary' variant
* **community-progress:** styled components is now a peer dependency





## [1.0.2](https://github.com/telusdigital/tds-community/compare/@tds/community-progress@1.0.1...@tds/community-progress@1.0.2) (2019-09-17)

**Note:** Version bump only for package @tds/community-progress





<a name="1.0.1"></a>
## [1.0.1](https://github.com/telusdigital/tds-community/compare/@tds/community-progress@1.0.0...@tds/community-progress@1.0.1) (2018-10-22)




**Note:** Version bump only for package @tds/community-progress
