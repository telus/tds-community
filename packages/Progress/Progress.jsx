import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { colorGainsboro, colorWhite } from '@tds/core-colours'
import { safeRest } from '@tds/util-helpers'

import Bar from './Bar/Bar'

const ProgressBarContainer = styled.div(({ size }) => ({
  display: 'inline-block',
  position: 'relative',
  background: colorWhite,
  width: '100%',
  height: size === 'mini' ? '12px' : '24px',
  borderRadius: size === 'mini' ? '12px' : '24px',
  borderWidth: 'thin',
  borderStyle: 'solid',
  borderColor: colorGainsboro,
  overflow: 'hidden',
  transform: 'translateZ(-99999999px)', // Fix for safari overflow bug
}))

/**
 * @version ./package.json
 */

const Progress = ({ size, children, ...rest }) => (
  <ProgressBarContainer {...safeRest(rest)} size={size}>
    {children}
  </ProgressBarContainer>
)

Progress.propTypes = {
  /**
   * The size of the progress bar where the default is 24px tall and mini is 12px tall.
   */
  size: PropTypes.oneOf(['default', 'mini']),
  children: PropTypes.element.isRequired,
}

Progress.defaultProps = {
  size: 'default',
}

Progress.Bar = Bar
export default Progress
