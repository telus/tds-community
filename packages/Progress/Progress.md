The Progress component is a wrapper-container for an number of Progress.Bar components.
