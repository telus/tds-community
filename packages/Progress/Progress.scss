@import '~@tds/core-colours/colours';
@import '~@tds/core-responsive/responsive';

.progressBarContainer {
  display: inline-block;
  position: relative;
  background: #ffff;
  width: 100%;
  border-width: thin;
  border-style: solid;
  border-color: $color-gainsboro;
  @extend %progress-bar-height;
  @extend %progress-bar-padding;
  @extend %progress-bar-radius;

  .progressBar {
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    overflow: hidden;
    height: 100%;
    @extend %progress-bar-radius;
    box-shadow: inset 0 -1px 1px rgba(255,255,255,0.3);

  }
  .primary {
    background-color: $color-accessible-green;
  }
  .secondary {
    background: repeating-linear-gradient(
      -60deg,
      #A9ACAF,
      #A9ACAF 2px,
      $color-gainsboro 1px,
      $color-gainsboro 3px
    );
  }
  .error {
    background-color: $color-cardinal
  }
  .disabled {
    background-color: $color-gainsboro
  }
}

%progress-bar-height {
  @include mq($until: md) {
    height: 20px;
  }
  @include mq($from: md ) {
    height: 30px;
  }
}

%progress-bar-padding {
  @include mq($until: md) {
    padding: 7.5px;
  }
  @include mq($from: md ) {
    padding: 12.5px;
  }
}
%progress-bar-radius {
  @include mq($until: md) {
    -moz-border-radius: 20px;
    -webkit-border-radius: 20px;
    border-radius: 20px;
  }
  @include mq($from: md ) {
    -moz-border-radius: 30px;
    -webkit-border-radius: 30px;
    border-radius: 30px;
  }
}
