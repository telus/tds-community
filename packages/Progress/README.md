# TDS Community: Progress
A progress indicator.  Can be used to indicated usage for example. 

## Breakpoints
Height of progress is 30px on medium size screens or greater, otherwise it is 20px.
