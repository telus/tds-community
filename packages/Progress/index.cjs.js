const Progress = require('./dist/index.cjs')

module.exports = Progress
