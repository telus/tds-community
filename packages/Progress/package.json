{
  "name": "@tds/community-progress",
  "version": "1.0.2",
  "description": "A progress indicator.  Can be used to indicated usage for example.",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds-community.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds-community/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "@tds/core-colours": "^2.2.1",
    "@tds/core-responsive": "^1.3.3",
    "@tds/util-helpers": "^1.2.0",
    "prop-types": "^15.6.2"
  }
}
