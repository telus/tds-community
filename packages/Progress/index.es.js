import Progress from './dist/index.es'

export default Progress
