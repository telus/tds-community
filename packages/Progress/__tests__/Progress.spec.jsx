import React from 'react'
import { shallow } from 'enzyme'

import Progress from '../Progress'

describe('Progress', () => {
  const doShallow = (props = {}) =>
    shallow(
      <Progress {...props}>
        <Progress.Bar percentage={0} />
      </Progress>
    )

  it('renders', () => {
    const progress = doShallow()

    expect(progress).toMatchSnapshot()
  })

  it('does other things', () => {
    const progress = doShallow()

    expect(progress).toExist()
  })

  it('passes additional attributes to the element', () => {
    const progress = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(progress).toHaveProp('id', 'the-id')
    expect(progress).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const progress = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(progress).not.toHaveProp('className', 'my-custom-class')
    expect(progress).not.toHaveProp('style')
  })
})
