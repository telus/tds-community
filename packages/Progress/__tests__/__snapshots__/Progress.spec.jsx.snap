// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders 1`] = `
<div
  className="progressBarContainer"
>
  <Bar
    percentage={0}
    variant="primary"
  />
</div>
`;
